huaweicloud/services/cce/resource_huaweicloud_cce_node_pool.go

```diff
@@ -645,30 +657,21 @@ func resourceCCENodePoolRead(_ context.Context, d *schema.ResourceData, meta int
 	}
 	mErr = multierror.Append(mErr, d.Set("data_volumes", volumes))
 
-	rootVolume := []map[string]interface{}{
-		{
-			"size":           s.Spec.NodeTemplate.RootVolume.Size,
-			"volumetype":     s.Spec.NodeTemplate.RootVolume.VolumeType,
-			"hw_passthrough": s.Spec.NodeTemplate.RootVolume.HwPassthrough,
-			"extend_params":  s.Spec.NodeTemplate.RootVolume.ExtendParam,
-			"extend_param":   "",
-		},
+	rootVolume := map[string]interface{}{
+		"size":           s.Spec.NodeTemplate.RootVolume.Size,
+		"volumetype":     s.Spec.NodeTemplate.RootVolume.VolumeType,
+		"hw_passthrough": s.Spec.NodeTemplate.RootVolume.HwPassthrough,
+		"extend_params":  s.Spec.NodeTemplate.RootVolume.ExtendParam,
+		"extend_param":   "",
 	}
 	if s.Spec.NodeTemplate.RootVolume.Metadata != nil {
-		rootVolume[0]["kms_key_id"] = s.Spec.NodeTemplate.RootVolume.Metadata.SystemCmkid
+		rootVolume["kms_key_id"] = s.Spec.NodeTemplate.RootVolume.Metadata.SystemCmkid
 	}
-	mErr = multierror.Append(mErr, d.Set("root_volume", rootVolume))
-
-	tagmap := utils.TagsToMap(s.Spec.NodeTemplate.UserTags)
-	mErr = multierror.Append(mErr,
-		d.Set("tags", tagmap),
-		d.Set("status", s.Status.Phase),
-	)
+	mErr = multierror.Append(mErr, d.Set("root_volume", []interface{}{rootVolume}))
```

Comment:
you can use flattenResourceNodeRootVolume

Reply:
done.